OpenSim/Common/Component.h

```diff
@@ -198,19 +198,16 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
     Component(const std::string& aFileName,
         bool aUpdateFromXMLNode = true) SWIG_DECLARE_EXCEPTION;
 
+    /** Construct Component as a subcomponent member of another Component.
+        Automatically marks the component as a subcomponent of the owner.
+        A subcomponent MUST be named. **/
+    explicit Component(const std::string& name, Component* owner=nullptr);
```

Comment:
There is something wrong with this constructor. From the comment I would have expected that the parent Component would be essential and that it would be passed as a writable reference. But it is optional and a pointer. I guess you intended this constructor to do double duty as some other kind of constructor but the comment doesn't describe that use. Probably there should be two distinct constructors something like this:
```c++
/** Create an empty Component base class object with a given name. */ // is this protected?
explicit Component(const std::string& name);
/** Create an empty Component base class object with a given name and make it a
subcomponent of a given parent Component. */
Component(const std::string& name, Component& parent);
```
That said, I still think the second constructor is awkward. Wouldn't this be more natural as an operation on the parent, something like `parent.adoptSubcomponent(Component* subcomponent);`?